t/apicast-policy-3scale-batcher.t

```diff
@@ -407,176 +356,148 @@ The caching policy will cache the first result and return it while backend is
-          }
-        }
-      }
-    }
-  })
-}
---- config
-  include $TEST_NGINX_APICAST_CONFIG;
-
-  location /transactions/oauth_authorize.xml {
-    content_by_lua_block {
-      local test_counter = ngx.shared.test_counter or 0
-      if test_counter == 0 then
-        ngx.shared.test_counter = test_counter + 1
-        ngx.exit(200)
-      else
-        ngx.log(ngx.ERR, 'auth should be cached but called backend anyway')
-        ngx.exit(502)
-      end
+--- configuration env eval
+use JSON qw(to_json);
+to_json({
+  services => [{
+    id => 42,
+    backend_version => 'oauth',
+    backend_authentication_type => 'service_token',
+    backend_authentication_value => 'token-value',
+    proxy => {
+      authentication_method => 'oidc',
+      oidc_issuer_endpoint => 'https://example.com/auth/realms/apicast',
+      api_backend => "http://test:$TEST_NGINX_SERVER_PORT/",
+      proxy_rules => [
+          { pattern => '/', http_method => 'GET', metric_system_name => 'hits', delta => 1  }
+      ],
+      policy_chain => [
+        {
+          name => "apicast.policy.3scale_batcher",
+          configuration => {}
+        },
+        {name => "apicast.policy.apicast"}
+      ]
     }
   }
-  location /transactions.xml {
-    content_by_lua_block {
-      ngx.req.read_body()
-      local post_args = ngx.req.get_post_args()
-      local app_id_match, usage_match
-      for k, v in pairs(post_args) do
-        if k == 'transactions[0][app_key]' then
-          ngx.exit(500)
-        elseif k == 'transactions[0][usage][hits]' then
-          usage_match = v == '2'
-        elseif k == 'transactions[0][app_id]' then
-          app_id_match = v == 'appid'
+  ],
+    oidc => [{
+    issuer => 'https://example.com/auth/realms/apicast',
+    config => { id_token_signing_alg_values_supported => [ 'RS256' ] },
+    keys => { somekid => { pem => $::public_key, alg => 'RS256' } },
+  }]
+});
+--- backend
+    location /transactions/oauth_authorize.xml {
+      content_by_lua_block {
+        local test_counter = ngx.shared.test_counter or 0
+        if test_counter == 0 then
+          ngx.shared.test_counter = test_counter + 1
+          ngx.exit(200)
+        else
+          ngx.log(ngx.ERR, 'auth should be cached but called backend anyway')
+          ngx.exit(502)
         end
-      end
-      ngx.shared.result = usage_match and app_id_match
+      }
     }
+    location /transactions.xml {
+      content_by_lua_block {
+        ngx.req.read_body()
+        local post_args = ngx.req.get_post_args()
+        local app_id_match, usage_match
+        for k, v in pairs(post_args) do
+          if k == 'transactions[0][app_key]' then
+            ngx.exit(500)
+          elseif k == 'transactions[0][usage][hits]' then
+            usage_match = v == '3'
+          elseif k == 'transactions[0][app_id]' then
+            app_id_match = v == 'appid'
+          end
+        end
+        ngx.shared.result = usage_match and app_id_match
+      }
```

Comment:
this is executed when `/force_report_to_backend` is called, when the backend_client is instantiated, so it includes the usage (hits) from `3` calls now instead of 2, because the `/force_report_to_backend` endpoint is an upstream too so it increases the hits as well. It used to be `2` in the old test because that endpoint was accessed without going through APIcast.